drivers/source/serial_io.c

```diff
@@ -64,9 +66,22 @@ obc_error_code_t sciPrintText(unsigned char *text, uint32_t length) {
 
     /* USER CODE BEGIN */
     // Print text to the serial port using sciSendBytes. Use the mutex to protect the SCI module.
+    
+        
+
+    if (xSemaphoreTake(mutex, UART_MUTEX_BLOCK_TIME) == pdTRUE) {
+        sciSendBytes(text, length);
+        xSemaphoreGive(mutex);
+    } else {
+        return OBC_ERR_CODE_MUTEX_CREATION_FAILED;
```

Comment:
There's a more appropriate error code for this. If we failed to take the mutex, that means we timed out. It doesn't necessarily mean the mutex wasn't created.